sr_wear_logger/src/sr_wear_logger.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python
+#
+# Copyright 2021 Shadow Robot Company Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the Free
+# Software Foundation version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import rospy
+import rospkg
+import os
+import yaml
+import roslaunch
+import requests
+import shutil
+import datetime
+from sensor_msgs.msg import JointState
+from sr_utilities_common.aws_manager import AWS_Manager
+
+
+class WearLogger():
+
+    def __init__(self):
+        self.aws_manager = AWS_Manager()
+        self._first_run = True
+        self._previous_values = {}
+        self._current_values = {}
+        self._threshold = 0.0175
+        self._hand_serial = rospy.get_param("~hand_serial")
+        self._log_file_path = rospkg.RosPack().get_path('sr_wear_logger') + "/" + self._hand_serial + "/"
+        self._log_file_name = "wear_data.yaml"
+        self._init_log()
+
+        if self._hand_serial != "":
+            rospy.Timer(rospy.Duration(2), self._save_data_localy)
+            rospy.Timer(rospy.Duration(10), self._upload_to_AWS)
+
+        rospy.Subscriber('/joint_states', JointState, self.callback)
+
```

Comment:
generally, for readability, it would be good if you try to place functions in order of usage as much as you can. e.g. first init_log(), then download_from_AWS() and so on. Sometimes it's hard but it helps the reader.

Reply:
Done